src/main/java/org/jenkinsci/plugins/durabletask/PowershellScript.java

```diff
@@ -0,0 +1,179 @@
+ * THE SOFTWARE.
+ */
+
+package org.jenkinsci.plugins.durabletask;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import hudson.EnvVars;
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.Plugin;
+import hudson.Launcher;
+import jenkins.model.Jenkins;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.io.File;
+import java.lang.Class;
+import java.lang.ClassLoader;
+import java.lang.reflect.Method;
+import java.lang.Exception;
+import java.lang.ClassNotFoundException;
+import java.lang.InstantiationException;
+import java.lang.NoSuchMethodException;
+import java.lang.IllegalAccessException;
+import java.lang.reflect.InvocationTargetException;
+import hudson.model.TaskListener;
+import java.io.IOException;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+/**
+ * Runs a Powershell script
+ */
+public final class PowershellScript extends FileMonitoringTask {
+    private final String script;
+    private boolean capturingOutput;
+    private boolean addVerboseOutput;
+
+    @DataBoundConstructor public PowershellScript(String script) {
+        this.script = script;
+    }
+    
+    public String getScript() {
+        return script;
+    }
+
+    @Override public void captureOutput() {
+        capturingOutput = true;
+    }
+
+    @SuppressFBWarnings(value="VA_FORMAT_STRING_USES_NEWLINE", justification="%n from master might be \\n")
+    @Override protected FileMonitoringController doLaunch(FilePath ws, Launcher launcher, TaskListener listener, EnvVars envVars) throws IOException, InterruptedException {
+        List<String> args = new ArrayList<String>();
+        PowershellController c = new PowershellController(ws);
+        
+        String cmd;
+        if (capturingOutput) {
+            // Using redirection in PowerShell produces extra newlines in output, so I am using [io.file]::WriteAllText to prevent corrupted output of exit code
+            cmd = String.format("& \"%s\" | Out-String > \"%s\" 2> \"%s\"; [io.file]::WriteAllText(\"%s\",$LastExitCode);", 
+                quote(c.getPowershellMainFile(ws)),
+                quote(c.getLogFile(ws)),
+                quote(c.getResultFile(ws)),
+                quote(c.getLogFile(ws)),
+                quote(c.getOutputFile(ws)));
+
+        } else {
+            cmd = String.format("& \"%s\" | Out-String > \"%s\" 2>&1; [io.file]::WriteAllText(\"%s\",$LastExitCode);",
+                quote(c.getPowershellMainFile(ws)),
+                quote(c.getResultFile(ws)),
+                quote(c.getLogFile(ws)));
+        }
+        
+        // Write the script and execution wrapper to powershell files in the workspace
+        c.getPowershellMainFile(ws).write(script + "\r\nexit 0", "UTF-8");
+        c.getPowershellWrapperFile(ws).write(cmd, "UTF-8");
+        
+        // Try to load the PowerShell plugin to produce the command line arguments for running the script
```

Comment:
You have a good point.  I was trying to be a good citizen by reusing some functionality from the powershell plugin.  But, I think we can safely remove the dependency and use the default params defined here as I don't believe the command line arguments for running a powershell script would change.